index/indexfile_windows.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"fmt"
+	"math"
+	"os"
+)
+
+// NewIndexFile returns a new index file. The index file takes
+// ownership of the passed in file, and may close it.
+func NewIndexFile(f *os.File) (IndexFile, error) {
+	return &indexFileFromOS{f}, nil
+}
+
+type indexFileFromOS struct {
+	f *os.File
+}
+
+func (f *indexFileFromOS) Read(off, sz uint32) ([]byte, error) {
+	r := make([]byte, sz)
```

Comment:
If we are really supporting Windows, we should use mmap here instead of direct reads. This provides the best experience, and I think it's an easier mental model for debugging/ support, to consistently use file mapping. Maybe we could re-introduce `mmap-go`, but just here for Windows? Or [implement it directly](https://github.com/edsrzf/mmap-go/blob/main/mmap_windows.go), doesn't look too bad :)